sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedProcessorTest.java

```diff
@@ -332,10 +315,136 @@ public void staledLeaseAcquiring() {
         receivedDocuments.clear();
     }
 
+    @Test(groups = { "simple" }, timeOut = 50 * CHANGE_FEED_PROCESSOR_TIMEOUT)
+    public void readFeedDocumentsAfterSplit() {
+        createdFeedCollectionForSplit = createFeedCollection(FEED_COLLECTION_THROUGHPUT_FOR_SPLIT);
+
+        // generate a first batch of documents
+        setupReadFeedDocuments(createdFeedCollectionForSplit, FEED_COUNT);
+
+        changeFeedProcessor = ChangeFeedProcessor.changeFeedProcessorBuilder()
+            .hostName(hostName)
+            .handleChanges(changeFeedProcessorHandler())
+            .feedContainer(createdFeedCollectionForSplit)
+            .leaseContainer(createdLeaseCollection)
+            .options(new ChangeFeedProcessorOptions()
+                .setLeasePrefix("TEST")
+                .setStartFromBeginning(true)
+                .setMaxItemCount(10)
+            )
+            .build();
+
+        changeFeedProcessor.start().subscribeOn(Schedulers.elastic())
+            .timeout(Duration.ofMillis(2 * CHANGE_FEED_PROCESSOR_TIMEOUT))
+            .onErrorResume(throwable -> {
+                log.error("Change feed processor did not start in the expected time", throwable);
+                return Mono.error(throwable);
+            })
+            .doOnSuccess(aVoid -> {
+                // Wait for the feed processor to receive and process the first batch of documents.
+                waitToReceiveDocuments(2 * CHANGE_FEED_PROCESSOR_TIMEOUT, FEED_COUNT);
+            })
+            .then(
+                // increase throughput to force a single partition collection to go through a split
+                createdFeedCollectionForSplit.readProvisionedThroughput().subscribeOn(Schedulers.elastic())
+                    .flatMap(currentThroughput ->
+                        createdFeedCollectionForSplit.replaceProvisionedThroughput(FEED_COLLECTION_THROUGHPUT).subscribeOn(Schedulers.elastic())
+                    )
+                .then()
+            )
+            .subscribe();
+
+        // Wait for the feed processor to receive and process the first batch of documents and apply throughput change.
+        try {
+            Thread.sleep(4 * CHANGE_FEED_PROCESSOR_TIMEOUT);
+        } catch (InterruptedException e) {
+            log.error(e.getMessage());
```

Comment:
don't catch `InterruptedException` if that's thrown, we can fail the test.

Reply:
We need to in this case, in order to avoid compilation errors...

Reply:
why can't we add exception to the test method signature?

Reply:
replied on a similar comment below...